infra/build_specified_commit.py

```diff
@@ -54,42 +59,116 @@ def build_fuzzer_from_commit(project_name,
       sanitizer=sanitizer,
       architecture=architecture,
       env_to_add=None,
-      source_path=old_repo_manager.repo_dir)
+      source_path=old_repo_manager.repo_dir,
+      mount_location=os.path.join('/src', old_repo_manager.repo_name))
+
+
+def run_command_in_image(image_name, command):
+  """Runs a specific command inside a docker image and returns the results.
+
+  Args:
+    image_name: The docker image for the command to be run in
+    command: The command to be run in the image
+
+  Returns:
+    the output of the command
+
+  Raises:
+    DockerExecutionError: on commands execution failing
+  """
+  command_to_run = [
+      'docker', 'run', '--rm', '-i', '--privileged', '-t', image_name
+  ]
+  command_to_run.extend(command)
+  print('Running command: %s' % command_to_run)
+  process = subprocess.Popen(command_to_run, stdout=subprocess.PIPE)
+  out, err = process.communicate()
+  if err:
+    raise DockerExecutionError('Error running command: %s with error: %s' %
+                               (command_to_run, err.decode('ascii')))
+  if out:
+    return out.decode('ascii'), process.returncode
+  return None, process.returncode
+
+
+def check_docker_for_commit(docker_image_name, dir_name, example_commit):
```

Comment:
we can simplify this a lot by putting this function into a script inside the docker base-builder image. as is this is a little difficult to follow and maintain.

Reply:
What part do you think should be run in docker? Identifying the repo with the commit?

Reply:
Done.